shared/src/main/scala/amf/core/client/scala/parse/document/Reference.scala

```diff
@@ -71,6 +71,6 @@ case class Reference(url: String, refs: Seq[RefContainer]) extends PlatformSecre
   def isInferred: Boolean = refs.exists(_.linkType == InferredLinkReference)
 }
 object Reference {
-  def apply(url: String, kind: ReferenceKind, node: YNode, fragment: Option[String]): Reference =
-    new Reference(url, Seq(RefContainer(kind, node, fragment)))
+  def apply(url: String, kind: ReferenceKind, pos: SourceLocation, fragment: Option[String]): Reference =
```

Comment:
Isn't it weird that we store the fragment separately? Storing the fragment is a very Json Schema specific thing

Reply:
On one side, this is just a refactor to encapsulate the fragment, but the concept was already present on the reference. On the other hand, is not just a json schema thing, other specs can reference thought fragments (path navigation)